x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1959,6 +1975,19 @@ private static boolean isNotQueryWithFromClauseAndFilterFoldedToFalse(UnaryPlan
             return (!(plan.child() instanceof LocalRelation) || (plan.child() instanceof LocalRelation &&
                 !(((LocalRelation) plan.child()).executable() instanceof EmptyExecutable)));
         }
+
+        private static LocalRelation findLocalRelation(LogicalPlan plan) {
+            Holder<LocalRelation> logicalPlanHolder = new Holder<>();
+            plan.transformDown(l -> {
```

Comment:
Is `transformDown` really needed here or `forEachDown` is enough?

Reply:
Ah, thx, didn't think of that!